### PR TITLE
Paintroid 433 - A zoomed-in view of what is being drawn

### PR DESCRIPTION
**Pull request by saat-sy**
An empty window appears when the canvas is clicked. The position of the window changes according to where the canvas is touched. 

https://jira.catrob.at/browse/PAINTROID-433

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Comment by ThorstenBandel** [comment]
Hey @saat-sy there seems to be one more error 

```
14:18:03  org.catrobat.paintroid.test.espresso.LandscapeIntegrationTest > testCorrectSelectionInBothOrientationsFillTool[android28(AVD) - 9] [31mFAILED [0m
14:18:03  	java.lang.NullPointerException: activity.findViewById(R.?.pocketpaint_zoom_window) must not be null
14:18:03  	at org.catrobat.paintroid.ui.zoomwindow.DefaultZoomWindowController.<init>(DefaultZoomWindowController.kt:94)
```
And a few conflicts. Would it be possible that you fix those things? :D

File: Paintroid/src/main/java/org/catrobat/paintroid/ui/zoomwindow/DefaultZoomWindowController.kt [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+                paint
+            ) }
+
+        return croppedBitmap
+    }
+
+    private fun mergeBackground(bitmap: Bitmap?): Bitmap? {
+
+        // Adding the extra width and height for the grey background
+        val bmOverlay =
```
Comment (ThorstenBandel) [inline Paintroid/src/main/java/org/catrobat/paintroid/ui/zoomwindow/DefaultZoomWindowController.kt:203]: please write out the variable names: bmOverlay => bitmapOverlay. 
This is for better readability of the code a few months down when somebody else needs to take a look at it

File: Paintroid/src/main/res/drawable/pocketpaint_zoom_window_shape.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android" android:shape="oval">
+    <stroke android:width="3dp" android:color="@color/pocketpaint_colorAccent"/>
+</shape>
```
Comment (ThorstenBandel) [inline Paintroid/src/main/res/drawable/pocketpaint_zoom_window_shape.xml:4]: Please add a newline at the end of the file

**Review by ThorstenBandel — CHANGES_REQUESTED** [review]
Looks good overall, one thing i saw is, that the zoom window does not change positions anymroe when beeing in the way, can you take a look at that? Other than that i just made some small remarks. Its mainly to remove some comments, as we try to not have any commenty, only if absolutetly necessary (comments tend to be written and then dont removed or changed if the corresponding code does also change)

**Review by ThorstenBandel — APPROVED** [review]
LGTM

**Comment by juliajulie95** [comment]
Looks fine, though I found a little bug :)
@ThorstenBandel I think we should still merge this and create a Bugticket for that.
If you pan to the right bottom corner of the image it is possible to draw under the zoom-window

https://user-images.githubusercontent.com/37836518/209829408-48837467-3cc3-4998-aca2-f3bca4986925.mp4

**Review by juliajulie95 — APPROVED** [review]

**Comment by juliajulie95** [comment]
Created a bug ticket: https://jira.catrob.at/browse/PAINTROID-524

**Pull request merged**